src/parameterizations/lateral/MOM_hor_visc.F90

```diff
@@ -468,81 +472,92 @@ subroutine horizontal_viscosity(u, v, h, diffu, diffv, MEKE, VarMix, G, GV, US,
+    do J=Jsq-2,Jeq+2 ; do I=Isq-2,Ieq+2
       dvdx_bt(I,J) = CS%DY_dxBu(I,J)*(vbtav(i+1,J)*G%IdyCv(i+1,J) &
                                     - vbtav(i,J)*G%IdyCv(i,J))
       dudy_bt(I,J) = CS%DX_dyBu(I,J)*(ubtav(I,j+1)*G%IdxCu(I,j+1) &
                                     - ubtav(I,j)*G%IdxCu(I,j))
     enddo ; enddo
 
-    call pass_vector(dudx_bt, dvdy_bt, G%Domain, stagger=BGRID_NE)
-    call pass_vector(dvdx_bt, dudy_bt, G%Domain, stagger=AGRID)
+    ! post barotropic tension and strain
+    if (CS%id_dudx_bt > 0) call post_data(CS%id_dudx_bt, dudx_bt, CS%diag)
+    if (CS%id_dvdy_bt > 0) call post_data(CS%id_dvdy_bt, dvdy_bt, CS%diag)
+    if (CS%id_dudy_bt > 0) call post_data(CS%id_dudy_bt, dudy_bt, CS%diag)
+    if (CS%id_dvdx_bt > 0) call post_data(CS%id_dvdx_bt, dvdx_bt, CS%diag)
 
     if (CS%no_slip) then
-      do J=js-1,Jeq ; do I=is-1,Ieq
+      do J=Jsq-2,Jeq+2 ; do I=Isq-2,Ieq+2
         sh_xy_bt(I,J) = (2.0-G%mask2dBu(I,J)) * ( dvdx_bt(I,J) + dudy_bt(I,J) )
       enddo ; enddo
     else
-      do J=js-1,Jeq ; do I=is-1,Ieq
+      do J=Jsq-2,Jeq+2 ; do I=Isq-2,Ieq+2
         sh_xy_bt(I,J) = G%mask2dBu(I,J) * ( dvdx_bt(I,J) + dudy_bt(I,J) )
       enddo ; enddo
     endif
 
     do j=Jsq-1,Jeq+2 ; do i=Isq-1,Ieq+2
-      grad_vel_mag_bt_h(i,j) = boundary_mask_h(i,j) * (dudx_bt(i,j)**2 + dvdy_bt(i,j)**2 + &
+      grad_vel_mag_bt_h(i,j) = G%mask2dT(I,J) * boundary_mask_h(i,j) * (dudx_bt(i,j)**2 + dvdy_bt(i,j)**2 + &
             (0.25*((dvdx_bt(I,J)+dvdx_bt(I-1,J-1))+(dvdx_bt(I,J-1)+dvdx_bt(I-1,J))))**2 + &
             (0.25*((dudy_bt(I,J)+dudy_bt(I-1,J-1))+(dudy_bt(I,J-1)+dudy_bt(I-1,J))))**2)
     enddo ; enddo
 
-    do J=js-2,Jeq+1 ; do I=is-2,Ieq+1
-      grad_vel_mag_bt_q(I,J) = boundary_mask_q(I,J) * (dvdx_bt(I,J)**2 + dudy_bt(I,J)**2 + &
+    do J=Jsq-2,Jeq+1 ; do I=Isq-2,Ieq+1
+      grad_vel_mag_bt_q(I,J) = G%mask2dBu(I,J) * boundary_mask_q(I,J) * (dvdx_bt(I,J)**2 + dudy_bt(I,J)**2 + &
             (0.25*((dudx_bt(i,j)+dudx_bt(i+1,j+1))+(dudx_bt(i,j+1)+dudx_bt(i+1,j))))**2 + &
             (0.25*((dvdy_bt(i,j)+dvdy_bt(i+1,j+1))+(dvdy_bt(i,j+1)+dvdy_bt(i+1,j))))**2)
     enddo ; enddo
 
-    do j=js-1,je+1 ; do i=is-1,ie+1
+    call pass_var(h, G%domain, halo=2)
+
+    do j=js-2,je+2 ; do i=is-2,ie+2
       htot(i,j) = 0.0
     enddo ; enddo
-    do k=1,nz ; do j=js-1,je+1 ; do i=is-1,ie+1
+    do k=1,nz ; do j=js-2,je+2 ; do i=is-2,ie+2
       htot(i,j) = htot(i,j) + GV%H_to_Z*h(i,j,k)
     enddo ; enddo ; enddo
 
     I_GME_h0 = 1.0 / CS%GME_h0
-    do j=Jsq,Jeq+1 ; do i=Isq,Ieq+1
+    do j=Jsq-1,Jeq+2 ; do i=Isq-1,Ieq+2
       if (grad_vel_mag_bt_h(i,j)>0) then
-        GME_effic_h(i,j) = CS%GME_efficiency * boundary_mask_h(i,j) * &
+        GME_effic_h(i,j) = CS%GME_efficiency * G%mask2dT(I,J) * &
             (MIN(htot(i,j) * I_GME_h0, 1.0)**2)
       else
         GME_effic_h(i,j) = 0.0
       endif
     enddo ; enddo
 
-    do J=js-1,Jeq ; do I=is-1,Ieq
+    do J=Jsq-2,Jeq+1 ; do I=Isq-2,Ieq+1
```

Comment:
This is another case where there should be symmetric halos based on the q-point computational domain, or non-symmetric halos based on the tracer point loop limits - i.e., `do J=Jsq-1,Jeq+1 ; do I=Isq-1,Ieq+1` or `do J=js-2,je+1 ; do I=is-2,ie+1`, depending on the later use of GME_effic_q.